### ISSUE TITLE
Fullscreen foes not work in i3

### ISSUE THREAD

**Issue by rumpelsepp** [eXeC64/imv#87]
In the i3 window manager the `f` key for fullscreen does not work. `imv` flickers and stays the same.

**Comment by czarkoff** [comment]
Can't confirm: for me pressing `f` key makes `imv` occupy full screen with `i3`. Please, provide more info:

* which versions of OS/distro, `sdl2`, `i3` and `imv` do you try?
* do you run X11 or Wayland?

**Comment by eXeC64** [comment]
I can confirm. I've got this bug with imv, but been too busy with academic
work to look into it. I'll do a bisection to find out when it was
introduced.

**Comment by rumpelsepp** [comment]
@czarkoff: 

```
$ i3 --version
i3 version 4.12 (2016-03-06, branch "4.12") © 2009 Michael Stapelberg and contributors

$ pacman -Qi sdl2
Name            : sdl2
Version         : 2.0.4-2
Description     : A library for portable low-level access to a video framebuffer, audio output, mouse, and keyboard (Version 2)
Architecture    : x86_64
URL             : http://www.libsdl.org
Licenses        : MIT
Groups          : None
Provides        : None
Depends On      : glibc  libxext  libxrender  libx11  libgl  libxcursor
Optional Deps   : alsa-lib: ALSA audio driver [installed]
                  libpulse: PulseAudio audio driver [installed]
Required By     : 0ad  imv  sdl2_ttf  xonotic
Optional For    : None
Conflicts With  : None
Replaces        : None
Installed Size  : 2,83 MiB
Packager        : Sven-Hendrik Haase <sh@lutzhaase.com>
Build Date      : Sa 02 Jan 2016 23:28:10 CET
Install Date    : So 13 Mär 2016 17:09:22 CET
Install Reason  : Installed as a dependency for another package
Install Script  : No
Validated By    : Signature

$ imv -h
imv v2.0.0
See manual for usage information.

Legal:
This program is free software; you can redistribute it and/or
modify it under the terms of the GNU General Public License
as published by the Free Software Foundation; either version 2
of the License, or (at your option) any later version.

This software uses the FreeImage open source image library.
See http://freeimage.sourceforge.net for details.
FreeImage is used under the GNU GPLv2.
```

Using this on arch linux, so everything should be up to date. i3 on X, no wayland started.

**Comment by eXeC64** [comment]
I bisected imv all the way back to the introduction of the full screen mode, and it's broken. So the problem is caused by something outside of vim.

I'm also on Arch linux with the exact same packages as @rumpelsepp.

@czarkoff What versions are you running?

**Comment by czarkoff** [comment]
$ pkg_info -aq | grep -E '(i3|sdl2)'
i3-4.12
i3status-2.10
sdl2-2.0.4p0
sdl2-ttf-2.0.13

I'll see whether there are any local patches for sdl2 or i3, but my guess is that it is an issue with sdl2 and X11 client libraries Arch has atm.

**Comment by czarkoff** [comment]
OK, I see the issue now. `f` works fine in stacking or tabbed mode and does not in split mode. That seems to happen because in split mode i3 strips `_NET_WM_STATE` from windows it manages. I'll look into this issue in more detail.

**Comment by czarkoff** [comment]
Apparently it is SDL's fault: for this particular key we recieve two `SDL_Event`s of type `SDL_KEYDOWN` and one of `SDL_KEYUP`. Possible fixes are:

1. Switch event handling to `SDL_KEYUP`: `imv` will only act when key is released.
2. Add state tracking for `SDL_KeyboardEvent`s: `imv` will remember the last key pressed and decide wether ignore its repetition or not.

The first option requires only a trivial fix (see the `keypress` branch), but it alters the way 'imv' responds to holding keys: instead of repeating action while key is pressed we just wait until user releases the key. This provides better timing control (people are quicker at releasing then pressing), but the price is a lot of key presses for resizing/panning images.

@eXeC64: The call is yours. If you like the first option, merge the branch. If you prefer the second option, remove the branch, and I'll draft a solution.

@rumpelsepp: please, test the branch.

**Comment by czarkoff** [comment]
OK, second option turned out to be not as difficult to implement as I have initially thought. Branch `keyrepeat` waits your decision @eXeC64.

@rumpelsepp, please test the branch.

**Comment by rumpelsepp** [comment]
keypress works; keyrepeat works as well. :+1:

**Comment by eXeC64** [comment]
I much prefer the the `keyrepeat` branch. It's nice and simple, and preserves existing behaviour. I'll merge `keyrepeat` and delete `keypress`.

Nice job! :+1:

**Comment by eXeC64** [comment]
Fixed by 9d50ed2
